plutus-core/plutus-core/src/PlutusCore/Builtin/Meaning.hs

```diff
@@ -71,11 +71,11 @@ class (Typeable uni, Typeable fun, Bounded fun, Enum fun, Ix fun) => ToBuiltinMe
     type CostingPart uni fun
 
     -- | Get the 'BuiltinMeaning' of a built-in function.
-    toBuiltinMeaning :: HasConstantIn uni val => fun -> BuiltinMeaning val (CostingPart uni fun)
+    toBuiltinMeaning :: HasConstantIn uni val => Version ann -> fun -> BuiltinMeaning val (CostingPart uni fun)
```

Comment:
We could move `Version` to the type level and add it as an argument to `ToBuiltinMeaning`, so that we don't need to bind it in each builtin. We could also index `DefaultFun` by `ver` and avoid extending `ToBuiltinMeaning` in any way at all. In both the cases we'd need to demote `ver` to a `Version` via a singleton or whatever, but that should be easy.

None of the options are probably worth it, though.